src/algebra/group/pi.lean

```diff
@@ -235,6 +236,21 @@ lemma pi.single_mul [Π i, mul_zero_class $ f i] (i : I) (x y : f i) :
   single i (x * y) = single i x * single i y :=
 (mul_hom.single f i).map_mul x y
 
+@[to_additive]
+lemma pi.mul_single_commute [Π i, mul_one_class $ f i] :
+  pairwise (λ i j, ∀ (x : f i) (y : f j), commute (mul_single i x) (mul_single j y)) :=
```

Comment:
Might be nice to also add
```lean
lemma commute.mul_single (x y : f i) (h : commute x y) : commute (mul_single i x) (mul_single i y) :=
```

Reply:
That'd be just `commute.map` with `momoid_hom.single`, right? Still with adding? But I guess yes, we have the other homomorphism properties of the unbundled `mul_single` as lemmas as well.

Reply:
Ah good point, then that's not all that helpful after all. Maybe just mention that in the docstring of this one; "for when i and j are equal but x and y commute, `commute.map` can be used" or similar